fifo_queue.go

```diff
@@ -219,3 +219,60 @@ func (st *FIFO) IsLocked() bool {
 
 	return st.isLocked
 }
+
+// MoveFrontWithId moves the element at index position to the front of the queue
+func (st *FIFO) MoveFrontWithId(index int) error {
+
+	if st.isLocked {
+		return NewQueueError(QueueErrorCodeLockedQueue, "The queue is locked")
+	}
+	st.rwmutex.Lock()
+	defer st.rwmutex.Unlock()
+
+	length := len(st.slice)
+	if length == 0 {
+		return NewQueueError(QueueErrorCodeEmptyQueue, "Empty queue")
+	}
+
+	if index == 0 {
+		return NewQueueError(QueueErrorCodeIndexFirstPosition, "Element already in first position")
+	}
+
+	if index > length {
+		return NewQueueError(QueueErrorCodeIndexOutOfBounds, "Index is out of bounds")
+	}
+
+	for i := index; i >= 1; i-- {
+		st.slice[i], st.slice[i-1] = st.slice[i-1], st.slice[i]
+	}
+
+	return nil
+}
+
+// MoveBackWithId moves the element at index position to the back of the queue
+func (st *FIFO) MoveBackWithId(index int) error {
+
+	if st.isLocked {
+		return NewQueueError(QueueErrorCodeLockedQueue, "The queue is locked")
+	}
+	st.rwmutex.Lock()
+	defer st.rwmutex.Unlock()
+
+	length := len(st.slice)
+	if length == 0 {
+		return NewQueueError(QueueErrorCodeEmptyQueue, "Empty queue")
+	}
+
+	if index == length-1 {
+		return NewQueueError(QueueErrorCodeIndexLastPosition, "Element already in last position")
+	}
+
+	if index > length {
```

Comment:
This should be `>=` because arrays are 0-based.

Please add two tests for this operation: one using length-1 as one of the positions and another using length as one of the positions. The first one should succeed and the second one should fail.